primo/utils/clustering_utils.py

```diff
@@ -15,17 +15,21 @@
 import logging
 
 # Installed libs
+import networkx as nx
```

Comment:
Is this included in requirements (pyproject.toml)?

Reply:
@DevKakkad32 See failing tests.